crates/codegen/src/verilog/memory.rs

```diff
@@ -0,0 +1,99 @@
+//! Transforms the graph to use memory / flip flops
+//! Replaces the root func node with loads from memory
+//! Replaces the leaf call nodes with stores to memory
+
+use std::collections::BTreeMap;
+
+use crate::*;
+use tohdl_ir::expr::*;
+use tohdl_ir::graph::DataFlow;
+use tohdl_ir::graph::*;
+use tohdl_passes::Transform;
+use tohdl_passes::TransformResultType;
+
+#[derive(Clone, PartialEq)]
+pub struct NextStateNode {
+    pub values: Vec<Expr>,
+}
+
+impl std::fmt::Display for NextStateNode {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        let buf = self
+            .values
+            .iter()
+            .map(|v| format!("{}", v))
+            .collect::<Vec<_>>()
+            .join(", ");
+        write!(f, "({})", buf)
+    }
+}
+
+impl DataFlow for NextStateNode {
+    fn referenced_vars(&self) -> Vec<&VarExpr> {
+        self.values.iter().flat_map(|v| v.get_vars_iter()).collect()
```

Comment:
Separate into multiple lines

Reply:
I actually don't use that struct. Removed.